sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/test/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStoreTests.java

```diff
@@ -0,0 +1,172 @@
+                assertEquals(EVENT_HUB_NAME, checkpointTest.getEventHubName());
+                assertEquals(CONSUMER_GROUP, checkpointTest.getConsumerGroup());
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void testListCheckpointsEmptyList() {
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers(PREFIX)).thenReturn(new HashSet<>());
+
+        StepVerifier.create(store.listCheckpoints(FULLY_QUALIFIED_NAMESPACE, EVENT_HUB_NAME, CONSUMER_GROUP))
+            .expectNextCount(0L)
+            .verifyComplete();
+    }
+
+    @Test
+    public void testCheckpointKeyNotStored() {
+
+        Set<String> value = new HashSet<>();
+        value.add(KEY);
+
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers(PREFIX)).thenReturn(value);
+        when(jedis.hmget(eq(KEY),
+            eq(JedisRedisCheckpointStore.CHECKPOINT))).thenReturn(null);
+
+        StepVerifier.create(store.listCheckpoints(FULLY_QUALIFIED_NAMESPACE, EVENT_HUB_NAME, CONSUMER_GROUP))
+            .expectError(IllegalStateException.class)
+            .verify();
+    }
+
+    @Test
+    public void testListOwnership() {
+        PartitionOwnership partitionOwnership = new PartitionOwnership()
+            .setFullyQualifiedNamespace(FULLY_QUALIFIED_NAMESPACE)
+            .setEventHubName(EVENT_HUB_NAME)
+            .setConsumerGroup(CONSUMER_GROUP)
+            .setPartitionId(PARTITION_ID)
+            .setOwnerId("ownerOne")
+            .setETag("eTag");
+
+        Set<String> value = new HashSet<>();
+        value.add(KEY);
+        byte[] bytes = jsonSerializer.serializeToBytes(partitionOwnership);
+        List<String> list = Collections.singletonList(new String(bytes, StandardCharsets.UTF_8));
+
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers(PREFIX)).thenReturn(value);
+        when(jedis.hmget(eq(KEY),
+            eq(JedisRedisCheckpointStore.PARTITION_OWNERSHIP))).thenReturn(list);
+
+        StepVerifier.create(store.listOwnership(FULLY_QUALIFIED_NAMESPACE, EVENT_HUB_NAME, CONSUMER_GROUP))
+            .assertNext(partitionOwnershipTest -> {
+                assertEquals(FULLY_QUALIFIED_NAMESPACE, partitionOwnershipTest.getFullyQualifiedNamespace());
+                assertEquals(EVENT_HUB_NAME, partitionOwnershipTest.getEventHubName());
+                assertEquals(CONSUMER_GROUP, partitionOwnershipTest.getConsumerGroup());
+                assertEquals("ownerOne", partitionOwnershipTest.getOwnerId());
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void testListOwnershipEmptyList() {
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers(PREFIX)).thenReturn(new HashSet<>());
+
+        StepVerifier.create(store.listOwnership(FULLY_QUALIFIED_NAMESPACE, EVENT_HUB_NAME, CONSUMER_GROUP))
+            .expectNextCount(0L)
```

Comment:
Why expect 0L? That means the mono is just expected to complete without emitting any items. you can remove this line.

Reply:
Thanks for explaining this. I went ahead and removed it :)